int32 front
int32 front_right
int32 front_left
int32 back